offline/coturn.md

```diff
@@ -0,0 +1,471 @@
+    iifname $INF_WAN ip daddr 5.9.84.121 udp dport 46884-61000 dnat to $SFTIP
+``` 
+
+This is the part that redirects the control traffic to the Coturn port:
+
+```nft
+    iifname $INF_WAN udp dport 3478 dnat to $COTURNIP:3478
+    iifname $INF_WAN tcp dport 3478 dnat to $COTURNIP:3478
+```
+
+
+Then we restart Nftables to apply the changes:
+
+```bash
+
+sudo systemctl restart nftables
+
+```
+
+## Change the Wire-Server configuration to use Coturn instead of Restund.
+
+We must change the Wire-Server configuration to use Coturn instead of Restund.
+
+First, we must locate what the "external" IP address of the machine is. 
+
+This is the IP we must provide in our Wire-Server configuration to allow the clients to connect to Coturn. 
+
+We get it by running the following command:
+
+```bash
+
+sudo ip addr
+
+```
+
+The first interface will be the loopback interface, `lo`, and the second interface will be the "external" interface, `enp41s0` in our example, the output looking something like this:
+
+```bash
+
+demo@install-docs:~/wire-server-deploy$ ip addr
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+    inet6 ::1/128 scope host 
+       valid_lft forever preferred_lft forever
+2: enp41s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
+    link/ether a8:a1:59:a2:9b:5b brd ff:ff:ff:ff:ff:ff
+    inet 5.9.84.121/32 scope global enp41s0
+       valid_lft forever preferred_lft forever
+    inet6 2a01:4f8:162:3b6::2/64 scope global 
+       valid_lft forever preferred_lft forever
+    inet6 fe80::aaa1:59ff:fea2:9b5b/64 scope link 
+       valid_lft forever preferred_lft forever
+3: etc... 
+
+``` 
+
+In this case, the external IP address is `5.9.84.121` so we will use that value for our Wire-Server configuration.
+
+Edit the `values/wire-server/values.yaml` file:
+
+```bash
+
+nano values/wire-server/values.yaml
+
+```
+
+You will find a section that looks like this (default):
+
+```yaml
+
+  turnStatic:
+    v1: []
+    v2:
+      # - "turn:<IP of restund1>:80"
+      # - "turn:<IP of restund2:80"
+      # - "turn:<IP of restund1>:80?transport=tcp"
+      # - "turn:<IP of restund2>:80?transport=tcp"
+      # - "turns:<IP of restund1>:443?transport=tcp"
+      # - "turns:<IP of restund2>:443?transport=tcp"
+
+``` 
+
+Or if you have already configured Restund, something like this:
+
+```yaml 
+
+  turnStatic:
+    v1: []
+    v2:
+      - "turn:<IP of restund1>:80"
+      - "turn:<IP of restund2>:80"
+      - "turn:<IP of restund1>:80?transport=tcp"
+      - "turn:<IP of restund2>:80?transport=tcp"
+
+``` 
+
+Instead, we configure it to use the external IP addres we found above, and the Coturn port, `3478` (as seen above in the `nftables` configuration):
```

Comment:
you have missed the part where we annotate each node with it's expected external IP.

Reply:
fixed in latest commit.

We must also annotate the node with the exrenal IP address we will be listening to (which we found with `sudo ip addr` above):

```bash

kubectl annotate node kubenode3 wire.com/external-ip='your.public.ip.address'

```